lua/mason-registry/init.lua

```diff
@@ -193,4 +193,23 @@ function M.refresh(cb)
     end
 end
 
+---table<string, string[]>
+local aliases = {}
+
+---Register aliases for the specified packages
+---@param new_aliases table<string, string[]>
+function M.register_aliases(new_aliases)
+    for key, value in pairs(new_aliases) do
+        aliases[key] = aliases[key] or {}
+        for _, alias in pairs(value) do
```

Comment:
The non-imperative way would perhaps be to add a new `_.merge_with` function to the functional modules, which would allow for something like:

```lua
function M.register_package_aliases(new_aliases)
    aliases = _.merge_with(_.concat, aliases, new_aliases)
end
```

I think this looks good to me though!

Reply:
Oh fantastic, definitely prefer to do this declaratively

Reply:
With the new feature of omitting aliases from the list if they match the mason package name it's a little wonky to do it declaratively now actually. Going to resolve this for now, happy to take another stab at it though if you think it's better